rf_tests.go

```diff
@@ -362,6 +369,108 @@ func uploadRFML(c cliContext, api rfmlAPI) error {
 	if err != nil {
 		return cli.NewExitError(err.Error(), 1)
 	}
+
+	wisp_tests, err := readWispFiles([]string{c.String("test-folder")})
+	if err != nil {
+		return cli.NewExitError(err.Error(), 1)
+	}
+	err = uploadWispFiles(wisp_tests, false, api)
+	if err != nil {
+		return cli.NewExitError(err.Error(), 1)
+	}
+
+	return nil
+}
+
+// uploadSingleWispFile uploads wisp file syntax by
+// trying to parse the file and sending any parse errors to the caller
+func uploadSingleWispFile(filePath string) error {
+	wisp, _ := readWispFile(filePath)
+
+	log.Printf("Updating existing wisp test: %v", wisp.TestID)
+	err := api.UpdateWisp(wisp)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func readWispFile(filePath string) (*rainforest.WispJson, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	wispReader := rainforest.NewWispReader(file)
+	wisp, err := wispReader.Read()
+	if err != nil {
+		return nil, fileParseError{filePath, err}
+	}
+
+	return wisp, err
+}
+
+// readWispFiles takes in a list of files and/or directories and a list of tags
+// and returns a list of the parsed tests, or an error if it is encountered. To
+// allow all tags, pass in nil for tags.
+func readWispFiles(files []string) ([]*rainforest.WispJson, error) {
+	fileList := []string{}
+	for _, file := range files {
+		stat, err := os.Stat(file)
+		if err != nil {
+			return nil, err
+		}
+		if !stat.IsDir() {
+			if strings.HasSuffix(file, ".json") {
+				fileList = append(fileList, file)
+				continue
+			} else {
+				log.Printf("%s is not a valid json file", file)
+				continue
+			}
+		}
+
+		// We have a directory, walk through and find wisp files
+		err = filepath.Walk(file, func(path string, f os.FileInfo, err error) error {
+			if strings.HasSuffix(path, ".json") {
```

Comment:
Bad stuff will happen if there's a sub-directory with a name of `foo.json`?

Reply:
I'll see how to check if something is a dir as well.

Reply:
Handling this is a bit involved since the way the code is structured now we would need to handle it in a few places.
Atm this blows up for both `rfml` and `wisp`. Wisp says that the json while is a dir, while rfml lists why the rfml is invalid.

I think wisp blows up relatively sensibly for now and if we want to handle this properly, we should probably restructure the code in a separate PR.